sources/pmwebapi.go

```diff
@@ -0,0 +1,164 @@
+package sources
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type GetToken struct {
+	Context int
+}
+
+type MetricHeader struct {
+	Metrics []PcpMetrics
+}
+
+type PcpMetrics struct {
+	Name        string
+	Pmid        int64
+	Indom       int64
+	TextOneline string `json:"text-oneline"`
+	TextHelp    string `json:"text-help"`
+	Sem         string
+	Units       string
+	Type        string
+}
+
+type TimestampHeader struct {
+	Timestamp
+	Values []GetValues
+}
+
+type Timestamp struct {
+	S  string
+	Us string
+}
+
+type GetValues struct {
+	Pmid      int64
+	Name      string
+	S         string
+	Us        string
+	Instances []InstanceList
+}
+
+type InstanceList struct {
+	Instance int64
+	Value    float64
+}
+
+type pcpPmwebapiMetric struct {
+	Labels      []string
+	Labelvalues []string
+	Name        string
+	TextHelp    string
+	Value       float64
+}
+
+func init() {
+	Factories["pmwebapi"] = newPcpSource
+}
+func gaugeMetric(labels []string, labelvalues []string, name string, textHelp string, value float64) prometheus.Metric {
+	return prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, "", name),
+			textHelp,
+			labels,
+			nil,
+		),
+		prometheus.GaugeValue,
+		value,
+		labelvalues...,
+	)
+}
+
+type pcpPmwebapiSource struct {
+	pcpPmwebapiMetrics []pcpPmwebapiMetric
+}
+
+func (p *pcpPmwebapiSource) Update(ch chan<- prometheus.Metric) error {
```

Comment:
Looking into this further, we will have to make several changes up the chain, including getting rid of the error checking on line 59 of `pcp_exporter.go`. I think it would be wise to keep this for now assuming we add error checking to this function when we add the more advanced label/type filtering.